### ISSUE TITLE
React to changes in NuGet 4.3.0

### ISSUE THREAD

**Issue by natemcmaster** [ericsink/SQLitePCL.raw#162]
Starting in NuGet 4.3.0, NuGet and the .NET SDK are adding a feature called "AssetTargetFallback". This will treat .NET Framework packages as compatible with .NET Core 2.0 and .NET Standard 2.0 (see the [full spec](https://github.com/NuGet/Home/wiki/Enable-.NET-Core-2.0-projects-to-work-with-.NET-Framework-4.6.1-compatible-packages).) In most cases, this will works correctly. However, there are edge cases, and SQLitePCLRaw is one of those. This edge case leads to the odd behavior discussed in https://github.com/dotnet/sdk/issues/1166. Current thinking is that https://github.com/dotnet/sdk/issues/1166 will be closed as "won't fix" because it only affects a very small number of packages currently on NuGet.org.

From initial inspection, it appears SQLitePCLRaw.lib.e_sqlite3.{linux, osx, v110_xp} and SQLitePCLRaw.lib.sqlcipher.{linux, osx, windows} need to make a few changes to work correctly in .NET Core 2.0 and .NET Standard 2.0 projects.

I'd recommend one of following approaches:

1. Move `build/net35/$pkgid.targets` back up to `build/$pkgid.targets`. Instead, use the following condition to exclude the targets from running in any .NET Standard and .NET Core projects.
   ```xml
   <Target Name="InjectReference_x" 
           Condition=" '$(TargetFrameworkIdentifier)' != '.NETCoreApp' AND '$(TargetFrameworkIdentifier)' != '.NETStandard' >
   ```
   cref https://github.com/dotnet/sdk/blob/a4bceb67b160c72318a581d87d52796e4fa31794/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets#L56-L60
2. Or, add "lib" assets to these packages to ensure NuGet doesn't import the `build/net35` targets file at all.
    ```
    + build/net35/$pkgid.targets
    - lib/
      + netstandard1.0/_._
      + netstandard2.0/_._
      + net35/_._
    ```
    The `netstandard1.0/_._` file ensures the package installs to .NET Core 1.0 and .NET Standard 1.x projects. 
    The `net35/_._` file ensures the package installs to .NET Framework. 
    The `netstandard2.0/_._` file ensures that when installed to .NET Core 2.0 or .NET Standard2.0, the `build/net35` targets are not imported.

cc @bricelam

@ericsink I'm happy to make a PR. The "AssetsTargetFallback" feature is still being implemented, so it may be a little while before we can test and verify any changes to SQLitePCL packages

**Comment by ericsink** [comment]
I'll take care of this.  My knee-jerk reaction is that I'll do approach (1), but I'll be giving this a closer look.

Thanks for the heads-up.

**Comment by natemcmaster** [comment]
FYI - we have a build of NuGet and .NET Core that have this `AssetTargetFallback` feature. You can download it here: https://github.com/dotnet/cli/tree/release/2.0.0#installers-and-binaries

With this new build, NuGet will now produce warnings about the 1.1.5 packages when used in .NET Core or .NET Standard 2.0.

```
 warning NU1701: Package 'SQLitePCLRaw.lib.e_sqlite3.linux 1.1.5' was restored using '.NETFramework,Version=v4.6.1' instead the project target framework '.NETCoreApp,Version=v2.0'. This may cause compatibility problems.
```

This warning will become a fatal error when users have set `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` (a fairly common setting).

**Comment by ericsink** [comment]
@natemcmaster Okay, I coded your suggestion (1), and then started second guessing myself, thinking maybe the other idea would be cleaner.  If you *were* going to do a PR for this, which of your two suggestions would you have implemented?

**Comment by natemcmaster** [comment]
(2) is cleaner and the one I would have made as PR. It's advantage is that it's behavior is more predictable. The presence of the  `_._` files tell NuGet about the packages' framework compatibility explicitly.

(1) is fine too. It has the advantage of being most similar to 1.1.3 (basically undoing #154 and taking a new approach). Fewer changes and staying on familiar ground are a nice too.

**Comment by ericsink** [comment]
Update:  I have reproduced the behavior with the new build of the .NET Core SDK, verifying that I do see that same warning with SQLitePCLRaw 1.1.5.  I have implemented suggestion (2), and verified that the warning disappears.

I plan to incorporate one or two other (unrelated) fixes and push up a 1.1.6 in the next few days.

**Comment by natemcmaster** [comment]
👏 great!

**Comment by ericsink** [comment]
Fixed in 1.1.6, which is now on nuget.org.  Feel free to let me know of any problems.

**Comment by natemcmaster** [comment]
Looks good, so far no issues. Thanks Eric.

**Comment by ericsink** [comment]
For reference, this issue was fixed in 1.1.6, but that release included SQLite 3.19.3, which contained a regression in LEFT JOIN, so EF core waited for 1.1.7, which included SQLite 3.18.2.
